CIP-0030/README.md

```diff
@@ -197,6 +197,16 @@ Errors: `APIError`, `PaginateError`
 
 If `amount` is `undefined`, this shall return a list of all UTXOs (unspent transaction outputs) controlled by the wallet. If `amount` is not `undefined`, this request shall be limited to just the UTXOs that are required to reach the combined ADA/multiasset value target specified in `amount`, and if this cannot be attained, `undefined` shall be returned. The results can be further paginated by `paginate` if it is not `undefined`.
 
+### api.getCollateralUtxos(amount: cbor\<Coin>): Promise\<TransactionUnspentOutput[] | null>
```

Comment:
Agreed about the positional parameters, can change this to be an object like `{ amount }`.

The `amount` must be a required parameter to incentivize dapps to not request more resources than they actually require. Depending on the specific wallet implementation a difference between asking for an abstract maximum collateral amount and the amount actually required at the moment might be the difference between the wallet responding immediately with already existing utxo and having to bother the user with a reorganisation when it's not critical.

Additionally, not specifying the required amount means the wallet have to just return whatever they have and then the dapp has to either use it all and rist additional unnecessary funds or have to implement some additional logic on how to select minimum amount form the returned utxos.

The name can be changed to `getCollateral` although it poses a question then if it's better for the existing daps to have a function that matches the spec by name, but not by parameters, or better to keep `getCollateral` as the experimental pre-spec name and then wallets can upgrade to support the function as described in the spec with the spec name

Reply:
I have updated the spec, renaming the function and changing the argument to an object. It now also comments that it is possible for wallets to not require the `amount` parameter, but it is not by-spec and is not recommended. This means the dapps should be working to be compatible with the spec as much as possible, and pass the required parameter, but the wallets are not forced to update, until they are ready.

Reply:
Think it's the best compromise solution that ensures the existing functions are kept backward compatible, but also allows to define the **right thing** in the spec for other wallets to have various options for the implementation and for dapps to follow.

Reply:
Yeah I would allow specifying the amount optionally. Right now wallets tend to return collateral in a wallet-dependent way, but usually everybody has some sort of arbitrarily-defined minimum (ex: 5 ADA minimum) based on the assumption that probably no dApp will require more than 5. It's not very scientific so it would be nice if the dApp could just specify how much it needs.